qiskit_experiments/database_service/db_analysis_result.py

```diff
@@ -149,24 +150,27 @@ def save(self) -> None:
                 "Analysis result cannot be saved because no experiment service is available."
             )
             return
-        # Get DB fit data
+
         value = self.value
         result_data = {
             "_value": value,
             "_extra": self.extra,
             "_source": self._source,
         }
 
-        # Display compatible float values in in DB
-        if isinstance(value, (int, float, bool)):
-            result_data["value"] = float(value)
-        elif isinstance(value, FitVal):
-            if isinstance(value.value, (int, float)):
-                result_data["value"] = value.value
+        # Format special DB display fields
+        if isinstance(value, FitVal):
+            db_value = self._display_format(value.value)
+            if db_value is not None:
+                result_data["value"] = db_value
             if isinstance(value.stderr, (int, float)):
-                result_data["variance"] = value.stderr ** 2
+                result_data["variance"] = self._display_format(value.stderr ** 2)
```

Comment:
I think you need to add a check for chi-sq, because if that's infinite, the API throws the same error as before.